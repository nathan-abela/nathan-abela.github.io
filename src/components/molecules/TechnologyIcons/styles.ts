import styled from 'styled-components';

const StyledTechnologyIcons = styled.div`
	.grid {
		display: grid;
		grid-template-columns: repeat(4, 1fr);
		gap: 1rem;

		@media screen and (max-width: 768px) {
			font-size: 32px;
		}

		@media screen and (max-width: 425px) {
			grid-template-columns: repeat(2, 1fr);
		}
	}

	.card {
		background-color: ${({ theme }) => theme.body};
		padding: 30px;
		cursor: pointer;
		border-radius: 12px;
		transition: all 0.35s ease-in-out 0s;
		-webkit-box-pack: start;
		justify-content: start;
	}

	.card:hover {
		// 0.2 opacity = 33
		box-shadow: ${({ theme }) => `0px 0px 50px ${theme.headline}33`};
	}

	.icon {
		display: flex;
		justify-content: center;
		color: ${({ theme }) => theme.headline};
	}

	.icon svg {
		height: 32px;
	}

	.icon a {
		color: ${({ theme }) => theme.headline};
	}

	// .tooltip {
	// 	position: relative;
	// 	display: inline-block;
	// }

	// .tooltip-text {
	// 	visibility: hidden;
	// 	width: 120px;
	// 	background-color: ${({ theme }) => theme.text};
	// 	color: ${({ theme }) => theme.body};
	// 	text-align: center;
	// 	padding: 4px 0;
	// 	border-radius: 6px;

	// 	// Tooltip Position
	// 	position: absolute;
	// 	top: -100%;
	// 	left: 50%;
	// 	transform: translateX(-50%);
	// 	opacity: 0;
	// 	transition: opacity 0.35s ease-in-out;
	// }

	// .icon:hover .tooltip-text {
	// 	visibility: visible;
	// 	opacity: 1;
	// }
`;

export { StyledTechnologyIcons };
