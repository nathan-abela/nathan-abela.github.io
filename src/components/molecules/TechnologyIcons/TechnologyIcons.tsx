import { FC } from 'react';
import { StyledTechnologyIcons } from './styles';

import { Icon } from 'Atoms/Icon';

const TechnologyIcons: FC = () => {
	return (
		<StyledTechnologyIcons>
			<div className="grid">
				<div className="card">
					<div className="icon tooltip">
						{/* <span className="tooltip-text">Javascript</span> */}
						<a
							href="https://www.javascript.com"
							target="_blank"
							rel="noopener noreferrer"
							title="JavaScript"
						>
							<Icon icon="JAVASCRIPT" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://www.typescriptlang.org"
							target="_blank"
							rel="noopener noreferrer"
							title="TypeScript"
						>
							<Icon icon="TYPESCRIPT" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://angular.io"
							target="_blank"
							rel="noopener noreferrer"
							title="Angular"
						>
							<Icon icon="ANGULAR" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://stenciljs.com"
							target="_blank"
							rel="noopener noreferrer"
							title="StencilJS"
						>
							<Icon icon="STENCILJS" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://nextjs.org"
							target="_blank"
							rel="noopener noreferrer"
							title="NextJS"
						>
							<Icon icon="NEXTJS" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://storybook.js.org"
							target="_blank"
							rel="noopener noreferrer"
							title="Storybook"
						>
							<Icon icon="STORYBOOK" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://www.python.org"
							target="_blank"
							rel="noopener noreferrer"
							title="Python"
						>
							<Icon icon="PYTHON" />
						</a>
					</div>
				</div>

				<div className="card">
					<div className="icon">
						<a
							href="https://playwright.dev"
							target="_blank"
							rel="noopener noreferrer"
							title="Playwright"
						>
							<Icon icon="PLAYWRIGHT" />
						</a>
					</div>
				</div>
			</div>
		</StyledTechnologyIcons>
	);
};

export { TechnologyIcons };
