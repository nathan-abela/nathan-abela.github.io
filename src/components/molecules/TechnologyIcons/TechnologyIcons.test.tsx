import { render } from '@testing-library/react';

import { TechnologyIcons } from './TechnologyIcons';

describe('Technology Icons', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<TechnologyIcons />
		);
		expect(container).toMatchSnapshot();
	});
});
