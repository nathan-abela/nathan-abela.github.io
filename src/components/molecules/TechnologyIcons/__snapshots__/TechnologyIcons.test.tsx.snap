// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology Icons renders correctly 1`] = `
<div>
  <div
    class="styles__StyledTechnologyIcons-sc-x4mqdq-0 ecLLGN"
  >
    <div
      class="grid"
    >
      <div
        class="card"
      >
        <div
          class="icon tooltip"
        >
          <a
            href="https://www.javascript.com"
            rel="noopener noreferrer"
            target="_blank"
            title="JavaScript"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://www.typescriptlang.org"
            rel="noopener noreferrer"
            target="_blank"
            title="TypeScript"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://angular.io"
            rel="noopener noreferrer"
            target="_blank"
            title="Angular"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://stenciljs.com"
            rel="noopener noreferrer"
            target="_blank"
            title="StencilJS"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://nextjs.org"
            rel="noopener noreferrer"
            target="_blank"
            title="NextJS"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://storybook.js.org"
            rel="noopener noreferrer"
            target="_blank"
            title="Storybook"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://www.python.org"
            rel="noopener noreferrer"
            target="_blank"
            title="Python"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="icon"
        >
          <a
            href="https://playwright.dev"
            rel="noopener noreferrer"
            target="_blank"
            title="Playwright"
          >
            <test-file-stub />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
